src/utils/selectors.ts

```diff
@@ -58,19 +59,33 @@ export const currentVenueSelectorData: SparkleSelector<AnyVenue | undefined> = (
  *
  * @param state the Redux store
  */
-export const worldUsersSelector: SparkleSelector<WithId<User>[] | undefined> = (
-  state
-) => state.firestore.ordered.worldUsers;
+export const worldUsersSelector: SparkleSelector<
+  WithId<UserWithLocation>[] | undefined
+> = (state) => state.firestore.ordered.worldUsers;
+
+export const worldUsersWithoutLocationSelector: SparkleSelector<
+  WithId<User>[] | undefined
+> = (state) => worldUsersSelector(state)?.map(userWithLocationToUser);
 
 /**
  * Selector to retrieve an object with world-related users from the Redux Firestore.
  *
  * @param state the Redux store
  */
 export const worldUsersByIdSelector: SparkleSelector<
-  Record<string, User> | undefined
+  Record<string, UserWithLocation> | undefined
 > = (state) => state.firestore.data.worldUsers;
 
+export const worldUsersByIdWithoutLocationSelector: SparkleSelector<
+  Record<string, WithId<User>> | undefined
+> = (state) => {
+  const worldUsersById = worldUsersByIdSelector(state);
+
+  return mapValues(worldUsersById, (user, userId) =>
+    userWithLocationToUser(withId(user, userId))
```

Comment:
Same here

Reply:
See my explanation to @denisdimitrov about this in https://github.com/sparkletown/sparkle/pull/1703#discussion_r664153565

it takes a `userWithLocation` and makes it a `user`.